### PR TITLE
Adds Node version and repo name to the cachedData.json

### PR DESCRIPTION
**Pull request by reteppeter**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -41,19 +41,22 @@ async function getRealNames(repoList: ReturnType<typeof getAllRepoDeps>, config:
 				.then(content => {
 					const name = content?.name || ""
 					const version = content?.version || ""
-					return { name: name,
-						 version: version
-						}
+					const languageVersion = content?._nodeVersion || undefined
```
Comment (mt-empty) [inline src/index.ts:44]: Why `undefined`, why not `null`?

File: src/index.ts [hunk 2]
```diff
@@ -87,11 +91,24 @@ async function getRealNames(repoList: ReturnType<typeof getAllRepoDeps>, config:
...
+			{
+				subPath: string,
+				subPathName: string,
+				blobPath: string,
+				version: string,
+				languageVersion?: string,
+				pushedAt: string,
+				dependencies: DependencyGraphDependency[],
+				realName: string
+			}[]
```
Comment (mt-empty) [inline src/index.ts:110]: Wouldn't it be better to define a type here

File: src/index.ts (hunk 1 above)
Comment (mt-empty) [reply src/index.ts:44]: Also down the road, if we want to store the value in JSON, then it won't work because JSON doesn't accept `undefined`

Comment (mt-empty) [inline src/index.ts:44]: why `_nodeVersion`, I though this would be `content.engines.node`?

**Review by mt-empty — CHANGES_REQUESTED** [review]
Well done

File: src/index.ts (hunk 1 above)
Comment (reteppeter) [reply src/index.ts:44]: Undefined mirrors the return value when a field isn't present at all. When we convert it to json (in depDataToJson), that field isn't present at all if it's undefined.

Comment (reteppeter) [reply src/index.ts:44]: It's untyped for me so I didn't realise there was another way, it's changed now.

Comment (mt-empty) [reply src/index.ts]: Ok sounds good, could you document it in the readme, so that its clear what the the crawler returns

**Review by mt-empty — APPROVED** [review]

File: src/graphQLAPI.test.ts [hunk 3]
```diff
@@ -83,7 +83,7 @@ test("test overall output of the crawler library", async () => {
 
 	const data = JSON.parse(await getJsonStructure(accessToken, {targetOrganisation: organisation}))
   if (data.npm.length > 0){
-    expect(Object.keys(data.npm[0]).length).toBe(280);
+    expect(Object.keys(data.npm[0]).length).toBe(282);
```
Comment (mt-empty) [inline src/graphQLAPI.test.ts:86]: How did this increase @reteppeter

**Pull request merged**